Code/Framework/AzFramework/Platform/Common/Xcb/AzFramework/XcbNativeWindow.cpp

```diff
@@ -219,17 +229,135 @@ namespace AzFramework
         }
     }
 
+    void XcbNativeWindow::DrawSplash()
+    {
+        const unsigned int bytesPerPixel = 4;
+        
+        typedef struct
+        {
+            const png_byte* data;
+            const png_size_t size;
+        } DataHandle;
+
+        typedef struct
+        {
+            const DataHandle data;
+            png_size_t offset;
+        } ReadDataHandle;
+
+        auto ReadDataFromInputStream = [](
+            png_structp png_ptr,
+            png_byte* raw_data,
+            png_size_t read_length)
+        {
+            ReadDataHandle* handle = (ReadDataHandle*)png_get_io_ptr(png_ptr);
+            const png_byte* png_src = handle->data.data + handle->offset;
+
+            memcpy(raw_data, png_src, read_length);
+            handle->offset += read_length;
+        };
+
+        png_structp png_ptr = png_create_read_struct(PNG_LIBPNG_VER_STRING, NULL, NULL, NULL);
+        png_infop info_ptr = png_create_info_struct(png_ptr);
+
+        ReadDataHandle handle = {{ logo_png, logo_png_len},0};
+
+        png_set_read_fn(png_ptr, &handle, ReadDataFromInputStream);
+        png_read_info(png_ptr, info_ptr);
+
+        int width = png_get_image_width(png_ptr, info_ptr);
+        int height = png_get_image_height(png_ptr, info_ptr);
+
+        png_bytep row_pointers[height];
```

Comment:
The height variable is not a constant and therefore can't be used as the size field of an array as that is a compile time calculation.
https://godbolt.org/z/bG7rrb1bf

However the rowPointers variable can use the array version of  `unique_ptr` per the C++ [Best Practices Guide](https://github.com/o3de/sig-core/blob/main/governance/C%2B%2B-Best-Practices-Guide.md#smart-pointers).
```suggestion
            const png_byte* png_src = handle->data.data + handle->offset;

            memcpy(raw_data, png_src, read_length);
            handle->offset += read_length;
        };

        png_structp png_ptr = png_create_read_struct(PNG_LIBPNG_VER_STRING, NULL, NULL, NULL);
        png_infop info_ptr = png_create_info_struct(png_ptr);

        ReadDataHandle handle = {{ logo_png, logo_png_len},0};

        png_set_read_fn(png_ptr, &handle, ReadDataFromInputStream);
        png_read_info(png_ptr, info_ptr);

        int width = png_get_image_width(png_ptr, info_ptr);
        int height = png_get_image_height(png_ptr, info_ptr);

        AZStd::unique_ptr<png_bytep[]> rowPointers;
```

If the allocated memory needs to be handed over to the `libpng` that can be done via using the [unique_ptr::release](https://en.cppreference.com/w/cpp/memory/unique_ptr/release) function, but looking at the code I don't see any place where a `libpng` deallocates the memory.

Reply:
Thank you for the help, changes applied